aws_lambda_builders/workflows/python_pip/packager.py

```diff
@@ -620,13 +639,79 @@ def _unpack_sdist_into_dir(self, sdist_path, unpack_dir):
+    def _get_name_version(self, pkg_info_filepath: str) -> Tuple[str, str]:
+        """
+        Extracts the name and the version from the PKG-INFO metadata file
+
+        Parameters
+        ----------
+        pkg_info_filepath: str
+            The path to the PKG-INFO file to get data from
+
+        Returns
+        -------
+        Tuple[str, str]
+            A tuple containing the name and version
+        """
+        metadata = self._parse_pkg_info_file(pkg_info_filepath)
+        return (metadata["Name"], metadata["Version"])
+
+    def _is_default_setuptools_values(self, name: str, version: str) -> bool:
+        """
+        Checks if the name or the version are the default values that are assigned by setuptools
+
+        Parameters
+        ----------
+        name: str
+            The name of the package
+        version: str
+            The version of the package as a string
+
+        Returns
+        -------
+        bool
+            True if either name or version are the default values
+        """
+        # default values logic:
+        # https://github.com/pypa/setuptools/blob/6083e18f4afc40316c0112134c205c336afbcdfd/setuptools/_distutils/dist.py#L1185-L1189
+        return name == "UNKNOWN" or version == "0.0.0"
+
+    def get_package_name_and_version(self, sdist_path: str) -> Tuple[str, str]:
+        """
+        Gets the package's name and version from the metadata file
+
+        Parameters
+        ----------
+        sdist_path: str
+            The string path of the downloaded source distribution artifact
+
+        Returns
+        -------
+        Tuple[str, str]
+            A tuple of the name and version of the package
+        """
         with self._osutils.tempdir() as tempdir:
             package_dir = self._unpack_sdist_into_dir(sdist_path, tempdir)
+
+            # get the name and version from the result setup.py
             pkg_info_filepath = self._get_pkg_info_filepath(package_dir)
-            metadata = self._parse_pkg_info_file(pkg_info_filepath)
-            name = metadata["Name"]
-            version = metadata["Version"]
+            name, version = self._get_name_version(pkg_info_filepath)
+
+            # return values if it is not the default values
+            if not self._is_default_setuptools_values(name, version):
+                return name, version
+
+            # see if we can get the fallback PKG_INFO file from the sdist
+            fallback_pkg_info_fp = self._get_fallback_pkg_info_filepath(package_dir)
+
+            if self._osutils.file_exists(fallback_pkg_info_fp):
+                # use the fallback values instead of the ones we generated
+                fallback_name, fallback_version = self._get_name_version(fallback_pkg_info_fp)
+
+                if not self._is_default_setuptools_values(fallback_name, fallback_version):
+                    name = fallback_name
+                    version = fallback_version
+
         return name, version
```

Comment:
in case they are not the fallback name and version, what are they?

Reply:
If the fallback name and version turn out to also be the default values, then it will use the the initially determined values. Basically, no replacement logic will happen.